compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -228,3 +235,137 @@ fn save_function_record(
     let is_used = true;
     coverageinfo::save_func_record_to_mod(cx, func_name_hash, func_record_val, is_used);
 }
+
+/// When finalizing the coverage map, `FunctionCoverage` only has the `CodeRegion`s and counters for
+/// the functions that went through codegen; such as public functions and "used" functions
+/// (functions referenced by other "used" or public items). Any other functions considered unused,
+/// or "Unreachable" were still parsed and processed through the MIR stage.
+///
+/// We can find the unreachable functions by the set different of all MIR `DefId`s (`tcx` query
+/// `mir_keys`) minus the codegenned `DefId`s (`tcx` query `collect_and_partition_mono_items`).
+///
+/// *HOWEVER* the codegenned `DefId`s are partitioned across multiple `CodegenUnit`s (CGUs), and
+/// this function is processing a `function_coverage_map` for the functions (`Instance`/`DefId`)
+/// allocated to only one of those CGUs. We must NOT inject any "Unreachable" functions's
+/// `CodeRegion`s more than once, so we have to pick which CGU's `function_coverage_map` to add
+/// each "Unreachable" function to.
```

Comment:
I understand your question, and in fact it's one of the examples that I just put in comment about why I'm adding each individual code region for `Unreachable`s, instead of one broad one.

If a function is covered when compiling one crate, but considered `Unreachable` from another, the overlapping but exactly equal code spans do not conflict. The `Unreachable` coverages just add `0` (no visible affect in coverage analysis).

I'll try to construct a test.

Reply:
Done. See new `uses_crate.rs` (with `deps/used_crate.rs`) test and results.